### ISSUE TITLE
Stackline disappears when not using stack for an amount of time

### ISSUE THREAD

**Issue by trankien386** [AdamWagner/stackline#74]
If I don't use stack for a long time and I start using stack again, stackline isn't visible for this time of use. I have to open Hammerspoon config and click Reload Config to make stackline reappear. It's very inconvenient!

**Comment by trankien386** [comment]
I've observed it closely and discovered how the bug happens. It seems like the bug only happens if you open a new window to make a stack with one current window which was the only window on the screen. When you add 1 more window to make it 1 stack-1 window side-by-side, stackline appears again.

I also discovered that if the new window to add to stack is from the same program as the current window, stackline appears perfectly. For example, you currently have one safari window filling the screen, you open a new safari window to make them stack, stackline appears as expected. But if you open one vscode window or one Terminal window to make them stack, stackline doesn't appear. And as I said above, opening any new window to make a side-by-side layout, stackline appears.

**Comment by AdamWagner** [comment]
Super helpful @trankien386 – thank you!

Will try to reproduce this

**Comment by trankien386** [comment]
> Super helpful @trankien386 – thank you!
> 
> Will try to reproduce this

Actually, it's more inconsistent than I thought. Right now I have only 1 Safari window filling the screen, and I just attempted to add a new Safari window on top of it to make it a stack, stackline didn't appear like before.

Did you ever experience this issue? Do you usually have to reload config on Hammerspoon Console periodically to make stackline show itself?

I think it has something to do with Hammerspoon.

**Comment by AdamWagner** [comment]
Unfortunately, no… I can't reproduce this.

There are some open issues on the Hammerspoon repo that suggest `WindowCreated` events are sometimes not fired. That could cause this.

https://user-images.githubusercontent.com/1683979/121452627-9ccc4280-c954-11eb-8062-cb93125037df.mp4

Last weekend, I added `WindowChanged` to the set of events that trigger a redraw. The `WindowChanged` event is a relatively new addition to the Hammerspoon window library.

Curious if you still see this issue running the latest version?

**Comment by trankien386** [comment]
I pulled the change you were referring to and for the last 2 days, I've forced myself to use stack whenever I could. I noticed that everything works perfectly, more than I expected. Not a single time that I use stack and stackline plays hide and seek with me, not anymore. No more Reload Config! It seems that God finally listens to my invocation.

That change is undoubtedly the step in the right direction. What a brilliant idea! Good work!   @AdamWagner.
